charts/synthetics-private-location/templates/secret.yaml

```diff
@@ -1,4 +1,4 @@
-{{- if and (not .Values.configConfigMap) (not .Values.configSecret) }}
+{{- if .Values.configFile }}
```

Comment:
This condition and the other ones must be `ne .Values.configFile "{}"` so that the default value does not create nor use the secret

https://github.com/DataDog/helm-charts/blob/0c9eb653dca3e96cb919a0503d8bd22b3ceaad48/charts/synthetics-private-location/values.yaml#L81-L82